pallets/msa/src/benchmarking.rs

```diff
@@ -196,64 +178,76 @@ benchmarks! {
 	}
 
 	revoke_delegation_by_delegator {
-		let (provider, provider_msa_id) = create_account_with_msa_id::<T>(0);
-		let (delegator, delegator_msa_id) = create_account_with_msa_id::<T>(1);
+		let provider_account = create_account::<T>("account", 0);
+		let (provider_msa_id, provider) = Msa::<T>::create_account(provider_account.into(), EMPTY_FUNCTION).unwrap();
+
+		let delegator_account = create_account::<T>("account", 1);
+		let (delegator_msa_id, delegator_public_key) = Msa::<T>::create_account(delegator_account.into(), EMPTY_FUNCTION).unwrap();
 
 		assert_ok!(Msa::<T>::add_provider(ProviderId(provider_msa_id), DelegatorId(delegator_msa_id), vec![]));
-	}: _ (RawOrigin::Signed(delegator), provider_msa_id)
+	}: _ (RawOrigin::Signed(delegator_public_key), provider_msa_id)
 	verify {
 		assert_eq!(frame_system::Pallet::<T>::events().len(), 1);
+		assert_eq!(Msa::<T>::get_delegation(DelegatorId(delegator_msa_id), ProviderId(provider_msa_id)).unwrap().revoked_at, T::BlockNumber::from(1u32));
 	}
 
 	create_provider {
-		let (provider, _provider_msa_id) = create_account_with_msa_id::<T>(1);
-	}: _ (RawOrigin::Signed(provider), Vec::from("Foo"))
+		let account = create_account::<T>("account", 0);
+		let (provider_msa_id, provider_public_key) = Msa::<T>::create_account(account.into(), EMPTY_FUNCTION).unwrap();
+	}: _ (RawOrigin::Signed(provider_public_key), Vec::from("Foo"))
+	verify {
+		assert!(Msa::<T>::get_provider_registry_entry(ProviderId(provider_msa_id)).is_some());
+	}
 
 	on_initialize {
```

Comment:
Since you are touching this here I think you should fix this. This benchmark runs very slowly and wastes a lot of time. I think we need to speed it up

Reply:
Yeah I think I am going to do this one in a different PR.